core/domain/collection_domain.py

```diff
@@ -906,133 +729,69 @@ def exploration_ids(self):
-                node: CollectionNode. A node in the exploration graph of a
-                    collection.
-
-            Returns:
-                list(str). A list of skill ids acquired by user.
-            """
-            if node.exploration_id in skills_learned_by_exp_id:
-                return skills_learned_by_exp_id[node.exploration_id]
-
-            skills_learned = set(node.acquired_skill_ids)
-            for other_node in self.nodes:
-                if other_node.exploration_id not in skills_learned_by_exp_id:
-                    for skill in node.prerequisite_skill_ids:
-                        if skill in other_node.acquired_skill_ids:
-                            skills_learned = skills_learned.union(
-                                _recursively_find_learned_skills(other_node))
-
-            skills_learned_by_exp_id[node.exploration_id] = skills_learned
-            return skills_learned
-
-        explorations_just_unlocked = []
-        explorations_qualified_for = []
-
-        collection_node = self.get_node(current_exploration_id)
-        collected_skills = _recursively_find_learned_skills(collection_node)
+        for index in range(0, len(self.nodes)-1):
+            if self.nodes[index].exploration_id == current_exploration_id:
+                exploration_just_unlocked = self.nodes[index+1].exploration_id
+                break
 
-        for node in self.nodes:
-            if node.exploration_id in skills_learned_by_exp_id:
-                continue
-
-            if set(node.prerequisite_skill_ids).issubset(set(collected_skills)):
-                if (any([
-                        skill in collection_node.acquired_skill_ids
-                        for skill in node.prerequisite_skill_ids])):
-                    explorations_just_unlocked.append(node.exploration_id)
-                else:
-                    explorations_qualified_for.append(node.exploration_id)
-
-        if explorations_just_unlocked:
-            return explorations_just_unlocked
-        elif explorations_qualified_for:
-            return explorations_qualified_for
-        else:
-            return []
+        return exploration_just_unlocked
 
     def get_nodes_in_playable_order(self):
-        """Returns a list of collection nodes in linear, playable order and
-        assumes the nodes can fit a linear structure.
+        """Returns a list of collection nodes in linear, playable order.
 
         Returns:
             list(CollectionNode). A sorted list of collection nodes.
         """
-        sorted_exp_ids = self.init_exploration_ids
-        next_exp_ids = self.get_next_exploration_ids(sorted_exp_ids)
-        while next_exp_ids:
-            for next_exp_id in next_exp_ids:
-                if next_exp_id not in sorted_exp_ids:
-                    sorted_exp_ids.append(next_exp_id)
-            next_exp_ids = self.get_next_exploration_ids(sorted_exp_ids)
+        sorted_exp_ids = []
+        if self.init_exploration_id:
```

Comment:
I also don't think I understand why we're doing it like this. Don't we already have a list of exploration nodes in order (i.e. self.nodes)?

Reply:
Done.